### ISSUE TITLE
BUG: `pycodestyle` in `pyproject.toml` even though the plugin is disabled through the config

### ISSUE THREAD

**Issue by hofrob** [flakeheaven/flakeheaven#149]
**Checklist**

* [X] Same issue occurs when reproducing the MWE below: new virtualenv, clean folder, minimal additional libs, etc.
* [X] Same issue occurs when running without cache (eg by setting the env var `FLAKEHEAVEN_CACHE_TIMEOUT=0`)

**Bug Description**

Disabling `pycodestyle` warnings for `pyproject.toml` does not work when adding the file as an argument in the cli. Enabling `pytest-mypy-plugins` to lint toml files is therefore not possible. If I understand this correctly, according to the docs it should be possible to enable/disable plugins for specific files/paths.

Add this config to `pyproject.toml` to not lint pycodestyle on this file:

```toml
[tool.flakeheaven.exceptions."pyproject.toml"]
pycodestyle = ["-*"]
```

Executing `flakeheaven lint pyproject.toml` show this warning:

```
pyproject.toml
     3:   2 E999 SyntaxError: cannot assign to expression here. Maybe you meant '==' instead of '='? [pycodestyle]
```

**Expected behavior**

No `pycodestyle` warnings should show up.

**Bug Repro**

In short (if you have [hatch](https://github.com/pypa/hatch) installed):

* execute `hatch new test-flakeheaven test-flakeheaven`
* add `flakeheaven` to `pyproject.toml` under `dependencies`
* add config to ignore `pycodestyle` in `pyproject.toml` (see below)
* execute `hatch run flakeheaven lint pyproject.toml`

1. Ubuntu 22.04
2. `python --version`: 3.10.6
3. `pyproject.toml` (and / or all other config files that apply, eg `.pylintrc`, stc)

```
[tool.flakeheaven]
exclude = [".git", "__pycache__", "venv/", "temp/"]
format = "grouped"

[tool.flakeheaven.exceptions."pyproject.toml"]
# or [tool.flakeheaven.exceptions."**/pyproject.toml"]
pycodestyle = ["-*"]
```
4.  `flakeheaven plugins`: pycodestyle and pyflakes (disabled)
5. Contents of any additional files (eg sample `.py` file(s) to be linted): just a normal toml file
6. [If available] contents of `flakeheaven config -vvv`

1. run `flakeheaven lint pyproject.toml`

My suspicion is that there's something wrong with the file/path matcher. If there's `tool.flakeheaven.exceptions` like so `**/foo.py`, it will only match files in subdirectories, not at the first/root level.

---

**This was an error on my side. Please ignore.**

Additional test case:

Add a file `foo.py` to root and a subdirectory with the content `a = 'bar'` (single quotes!). Add an exception to pyproject.toml:

```
[tool.flakeheaven.exceptions."**/foo.py"]
flake8-black = ["-*"]
```

`flakeheaven` will still report:

```
foo.py
     1:   5 BLK100 Black would make changes. [flake8-black]
```

But nothing on `subdir/foo.py`.

**Comment by pwoolvett** [comment]
Im checking to understand whats going on.

The additional test case works as expected:

```shell
$ FLAKEHEAVEN_CACHE_TIMEOUT=0 flakeheaven lint subdir/foo.py
(no output, exit code zero)
```
then, after commenting out the `flake8-black` exception:
```shell
$ FLAKEHEAVEN_CACHE_TIMEOUT=0 flakeheaven lint subdir/foo.py 

subdir/foo.py
     1:   5 BLK100 Black would make changes. [flake8-black]
```

**Comment by hofrob** [comment]
@pwoolvett You're right, thanks for checking! It works when I use the env var to disable cache.

I tried a few more testcases for the original problem, but still can't get it to work. Whatever pattern I use, `lint` always reports this one error. Patterns I tried:

* `**.toml`
* `*.toml`
* `**/*.toml`
* `pyproject.toml`
* `*/**.toml`
* `**/pyproject.toml`

**Comment by pwoolvett** [comment]
There's a difference in flakeheaven between actually running the plugin and running it but not report some exceptions.

* two errors are targeted to be always reported, so even if ignored specifically they will be produced: https://github.com/flakeheaven/flakeheaven/blob/a981c8cd5617bc4691009c998e54f5aa3a4bbbd0/flakeheaven/_logic/_plugin.py#L100

* we could skip processing the file altogether with specific plugin. For example, when deciding whether to process a file: https://github.com/flakeheaven/flakeheaven/blob/a981c8cd5617bc4691009c998e54f5aa3a4bbbd0/flakeheaven/_patched/_checkers.py#L119 we're comparing the rules like this `set(rules) == {'-*'}` (which is debatable, to say the least).

Instead, we could do `if '-*' in rules`. In this case, enabling a plugin globally in the plugins section (eg `pycodestyle = ['+*']`), and then disabling its execution explicitly in the exceptions with `pycodestyle = ['-*']` would avoid even processing the file

**Comment by hofrob** [comment]
Without knowing much about the internals, I think a simple `"-*" in rules` check won't be enough since you can have a config that looks like this:

```toml
# enable only codes from S100 to S199
flake8-bandit = ["-*", "+S1??"]
```

Taken from the [docs](https://flakeheaven.readthedocs.io/en/latest/config.html#example).

**Comment by pwoolvett** [comment]
hi @hofrob this should be fixed now and released as [3.2.1](https://pypi.org/project/flakeheaven/3.2.1/)
